changelog/unreleased/issue-4529

```diff
@@ -0,0 +1,14 @@
+Enhancement: Verify data integrity before upload
+
+Hardware issues or a bug in restic could cause restic to create corrupted files
+that were then uploaded to the repository. Detecting such corruption usually
+required explicitly running the `check --read-data` command.
+
+To prevent the upload of corrupted data to the repository, restic now
+additionally verifies that files can be decoded and contain the correct data
+beforehand. This increases the CPU usage during backups. If absolutely
+necessary, you can disable the verification using the option
+`--no-extra-verify`.
+
+https://github.com/restic/restic/issues/4529
+https://github.com/restic/restic/pull/4681
```

Comment:
Suggest the following:

~~~
Enhancement: Add extra verification of data integrity before upload

Hardware issues, or a bug in restic or its dependencies, could previously cause
corruption in the files restic created and stored in the repository. Detecting
such corruption previously required explicitly running the `check --read-data`
or `check --read-data-subset` commands.

To further ensure data integrity, even in the case of hardware or software bugs,
restic now performs additional verification of the files about to be uploaded
to the repository.

These extra checks will increase CPU usage during backups. They can therefore,
if absolutely necessary, be disabled using the `--no-extra-verify` option of
the `backup` command. Please note that this should be combined with more active
checking using the previously mentioned check commands.

https://github.com/restic/restic/issues/4529
https://github.com/restic/restic/pull/4681
~~~